### PR TITLE
Add paymaster demo

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**

Add paymaster demo
- Add paymaster demo main component
- Fix overlapping header

Screenshot 2024-05-22 at 4 39 09 PM

Screenshot 2024-05-22 at 11 17 18 AM

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: template/web/app/api/paymasterProxy/route.test.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { NextRequest } from 'next/server';
 import { paymasterClient } from '@/utils/paymasterClient';
 import { POST } from './route';
 
-jest.mock('@/utils/paymasterClient');
+jest.mock('../../../src/utils/paymasterClient');
```
[the remark this answers is not in the document]
Comment (Zizzamia) [reply template/web/app/api/paymasterProxy/route.test.ts:5]: @kyhyco thoughts on this?

File: template/web/app/paymaster-bundler/_components/CallStatus.tsx [hunk 2]
```diff
@@ -26,5 +26,10 @@ export function CallStatus({ id }: CallStatusProps): JSX.Element {
     },
   }) as UseCallsStatusReturnType;
 
-  return <div>Status: {callsStatus?.status ?? 'loading'}</div>;
+  return (
+    <div>
+      <br />
+      <strong>Status:</strong> {callsStatus?.status ?? 'loading'}
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (Zizzamia) [reply template/web/app/paymaster-bundler/_components/CallStatus.tsx:34]: Niiice

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import { useWriteContracts } from 'wagmi/experimental';
+import isLocal from '../../../src/utils/isLocal';
+import { usePaymasterBundlerContract } from '../_contracts/usePaymasterBundlerContract';
+import { CallStatus } from './CallStatus';
+
+// Use the local API URL to target the Paymaster directly without a proxy
+// if running on localhost, otherwise use the Paymaster Proxy.
+const paymasterURL = process.env.PAYMASTER_URL;
+const isLocalEnv = isLocal();
+const defaultUrl = isLocalEnv ? paymasterURL : `${document.location.origin}/api/paymaster-proxy`;
```
Comment (Zizzamia) [inline template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:12]: Follow up PR, I feel like we can build a getDefaultURL

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  const contract = usePaymasterBundlerContract();
+
+  if (contract.status !== 'ready') {
+    console.error('Contract is not ready');
+    return null;
+  }
+
+  const handleMint = () => {
+    if (!account.address) {
+      alert('You need to be signed in to mint an NFT.');
```
Comment (Zizzamia) [inline template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:27]: Why an `alert`? it feels so 2004

What are we trying to do here? Can we do it in a way it's more 2024.

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        paymasterService: {
+          url: defaultUrl,
+        },
+      },
+    });
+  };
+
+  return (
+    <div
+      className={clsx('w-full', 'flex', 'flex-col', 'items-center', 'justify-center', 'text-white')}
```
Comment (Zizzamia) [inline template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:51]: I believe clsx does not need a `,` for every classname, but you can use the `,` to just help group of them.

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            'text-white',
+            'hover:bg-blue-700',
+            'transition',
+            'duration-300',
+          )}
+          onClick={handleMint}
+        >
+          Mint NFT
+        </button>
+        {id && <CallStatus id={id} />}
```
Comment (Zizzamia) [inline template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:113]: what's `id`? can we be more explicit?

Comment (cpcramer) [reply template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx]: Updated `id` to be `transactionID`

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx (hunk 5 above)
Comment (cpcramer) [reply template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:51]: I actually get an error when I try to remove the `,`. 

`',' expected.ts(1005)
`

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx (hunk 4 above)
Comment (cpcramer) [reply template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx]: Updated to use react-toastify - much better and customizable

Comment (cpcramer) [reply template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx]: Screenshot 2024-05-22 at 3 18 50 PM

File: template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx (hunk 5 above)
Comment (Zizzamia) [reply template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:45]: you get error when you do thinks like this https://github.com/coinbase/build-onchain-apps/blob/59b3e0bfa4e11c717d84034956c633f408e7cabc/template/web/src/components/Button/Button.tsx#L31 ?

Comment (cpcramer) [reply template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx:45]: Updating

**Review by Zizzamia — APPROVED** [review]
Amazing

**Pull request merged**